util/ranger/ranger_test.go

```diff
@@ -259,6 +259,18 @@ func (s *testRangerSuite) TestTableRange(c *C) {
 			filterConds: "[or(or(or(eq(test.t.a, 1), eq(test.t.a, 3)), eq(test.t.a, 4)), and(gt(test.t.b, 1), or(eq(test.t.a, -1), eq(test.t.a, 5))))]",
 			resultStr:   "[[-1,-1] [1,1] [3,3] [4,4] [5,5]]",
 		},
+		{
+			exprStr:     "a in (1, 1, 1, 1, 1, 1, 2, 1, 2, 3, 2, 3, 4, 4, 1, 2)",
+			accessConds: "[in(test.t.a, 1, 1, 1, 1, 1, 1, 2, 1, 2, 3, 2, 3, 4, 4, 1, 2)]",
+			filterConds: "[]",
+			resultStr:   "[[1,1] [2,2] [3,3] [4,4]]",
+		},
+		{
+			exprStr:     "a not in (1, 2, 3)",
+			accessConds: "[not(in(test.t.a, 1, 2, 3))]",
+			filterConds: "[]",
+			resultStr:   "[[-9223372036854775807,0] [4,+inf)]",
```

Comment:
Why not `[-inf,0]`?

Reply:
and it shouldn't be -9223372036854775807 either way, most possibly a bug?